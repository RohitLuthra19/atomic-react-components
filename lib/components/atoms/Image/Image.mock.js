const defaultImage = {
  src: '/images/dummies/320.png',
  srcset:
    '/images/dummies/1280.png 1024w, /images/dummies/640.png 640w, /images/dummies/320.png 320w',
  sizes: '(min-width: 36em) 33.3vw, 100vw',
  placeholderSrc: '/images/image-placeholder.png',
  alt: 'Dummy Image',
};

export const lazyLoadedImage = {
  ...defaultImage,
  loading: 'lazy',
};

export default defaultImage;
