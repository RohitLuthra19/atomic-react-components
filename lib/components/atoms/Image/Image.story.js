// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';

import defaultImage, { lazyLoadedImage } from './Image.mock';
import Image, { ImageVanilla } from '.';

// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('Atoms', module).addWithChapters('Image', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => <ImageVanilla {...defaultImage} className="hide-default-sample" />,
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Responsive Image Variations',
      sections: [
        {
          title: 'Default',
          sectionFn: () => {
            console.log(defaultImage);
            return <Image {...defaultImage} />;
          },
        },
      ],
    },
    {
      title: 'Lazy loaded Image Variations',
      sections: [
        {
          title: 'Default lazy loaded',
          sectionFn: () => <Image {...lazyLoadedImage} />,
        },
      ],
    },
  ],
});
