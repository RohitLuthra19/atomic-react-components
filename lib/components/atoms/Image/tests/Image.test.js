import React from 'react';
import { shallow, render } from 'enzyme';

import { ImageVanilla } from '../Image';

describe('Image Component', () => {
  const sampleProps = {
    src: 'http://abc.com',
    placeholderSrc: 'http://xyz.com',
    alt: 'ABC',
    elementLocator: 'my-custom-image-locator',
  };

  test('should render correctly and match snapshots', () => {
    const image = shallow(<ImageVanilla {...sampleProps} />);
    expect(image).toMatchSnapshot();
  });

  test('should change image src on error', () => {
    const image = shallow(<ImageVanilla {...sampleProps} />);
    image.simulate('error', { target: image });
    expect(image.src).toBe(sampleProps.placeholderSrc);
  });
  test('should check the data attribute of image', () => {
    const image = render(<ImageVanilla {...sampleProps} />);

    expect(image.attr('data-locator')).toEqual(sampleProps.elementLocator);
  });
});
