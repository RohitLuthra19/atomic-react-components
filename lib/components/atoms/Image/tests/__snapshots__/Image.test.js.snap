// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component should render correctly and match snapshots 1`] = `
<img
  alt="ABC"
  onError={[Function]}
  sizes={null}
  src="http://abc.com"
  srcSet={null}
/>
`;
